### ISSUE TITLE
publish tags

### ISSUE THREAD

**Issue by glensc** [danger/danger-gitlab-gem#5]
this repo has gems released but none of them tagged:

https://github.com/danger/danger-gitlab-gem/releases

**Comment by manicmaniac** [comment]
Could be too heuristic but the following script finds the latest commit at the time each version was published.

```ruby
require 'json'
require 'open-uri'

versions = JSON.parse(URI.open('https://rubygems.org/api/v1/versions/danger-gitlab.json').read)
versions
  .each do |version|
    sha, committer_date = `git log -n 1 --before #{version['created_at']} --pretty='format:%H %cd' --date=iso-strict-local`.split(' ')
    committed_at = Time.parse(committer_date).utc
    published_at = Time.parse(version['created_at']).utc
    puts <<~EOF
      ### #{version['number']} #{sha}
      - Committed at #{committed_at}
      - Published at #{published_at}
    EOF
  end
```

> ### 8.0.0 2f3cf2f43d012f2a3d1a9506fc6ea3eab763f3a3
> - Committed at 2020-05-19 12:40:06 UTC
> - Published at 2020-05-19 12:45:34 UTC
> ### 7.0.0 47ed0471737490d2f87d97359de88e728da8ec0b
> - Committed at 2019-03-12 16:36:39 UTC
> - Published at 2019-03-12 16:38:49 UTC
> ### 6.0.0 f96b6d6e70148da723780473684474751a3b65e1
> - Committed at 2017-07-25 11:49:42 UTC
> - Published at 2017-07-25 11:49:57 UTC
> ### 5.0.1 6086069eb56f45daef7ff8853c0089780f49486e
> - Committed at 2017-04-21 17:18:59 UTC
> - Published at 2017-04-21 17:20:31 UTC
> ### 5.0.0 64e6a6db9b610d3af581bb9b9a2a6339061f3e52
> - Committed at 2017-04-21 17:05:06 UTC
> - Published at 2017-04-21 17:05:35 UTC
> ### 4.3.1 48b0495b8448ef33e446a128a496f67445c624de
> - Committed at 2017-04-10 13:02:44 UTC
> - Published at 2017-04-10 13:03:11 UTC
> ### 4.0.1 dfb0e788cbb53c70c0a20f451e7f5edd409f342d
> - Committed at 2016-11-02 16:35:54 UTC
> - Published at 2016-11-02 16:36:32 UTC
> ### 4.0.0 de34a9d445590d542496dc3061dd6c8ab157137c
> - Committed at 2016-10-26 09:39:02 UTC
> - Published at 2016-10-26 09:39:22 UTC
> ### 1.0.0 637274e9bff8ec60d4fb1bcc7923db282d967dba
> - Committed at 2016-10-26 09:04:02 UTC
> - Published at 2016-10-26 09:12:41 UTC

It looks precise enough :eyes:

**Comment by manicmaniac** [comment]
I just thoroughly checked every commit and version and have confidence that https://github.com/danger/danger-gitlab-gem/issues/5#issuecomment-1408640213 is correct except for `1.0.0`.

`1.0.0` would be 296ca15 because https://rubygems.org/gems/danger-gitlab/versions/1.0.0 says it requires `gitlab ~> 3.7` and it was changed at 296ca15.

**Comment by manicmaniac** [comment]
I just pushed tags. Let me close this issue.

https://github.com/danger/danger-gitlab-gem/tags


I used the following script to add tags (and fixed `1.0.0` manually).

```ruby
require 'json'
require 'open-uri'

versions = JSON.parse(URI.open('https://rubygems.org/api/v1/versions/danger-gitlab.json').read)
versions
  .each do |version|
    sha, committer_date = `git log -n 1 --before #{version['created_at']} --pretty='format:%H %cd' --date=iso-strict-local`.split(' ')
    committed_at = Time.parse(committer_date).utc
    published_at = Time.parse(version['created_at']).utc
    number = version['number']
    puts <<~EOF
      ### #{version['number']} #{sha}
      - Committed at #{committed_at}
      - Published at #{published_at}
    EOF
    # Tag style (name and message) is same as `rake release` does.
    `git tag -am 'Version #{number}' v#{number} #{sha}`
  end
```
